src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java

```diff
@@ -27,14 +33,28 @@ public WebSecurityConfig(final UserDetailsService userDetailsService,
         this.encoder = encoder;
     }
 
+    @Bean
+    public AccessDeniedHandler accessDeniedHandler() {
+        return new CustomAccessDeniedHandler();
+    }
+
+    @Bean
+    RestAuthenticationEntryPoint authenticationEntryPoint() {
+        return new RestAuthenticationEntryPoint();
+    }
+
     @Override
     protected void configure(final AuthenticationManagerBuilder authenticationManagerBuilder) throws Exception {
         authenticationManagerBuilder.userDetailsService(userDetailsService).passwordEncoder(encoder);
     }
     @Override
     protected void configure(final HttpSecurity http) throws Exception {
         http
+                .exceptionHandling().accessDeniedHandler(accessDeniedHandler())
+
+                .and()
                 .httpBasic()
+                .authenticationEntryPoint(authenticationEntryPoint())
```

Comment:
When is it used?

Reply:
this method handles UnauthorizedAccessException